File: cassidy/python_code/topic.py
from kafka import KafkaAdminClient
from kafka.admin import NewTopic


def main():
    try:
        admin = KafkaAdminClient(
            bootstrap_servers=['localhost:9093', 'localhost:9094', 'localhost:9095'])
        print("Connected")

        topic = NewTopic(name='topic', num_partitions=3, replication_factor=3)
        topic2 = NewTopic(name='topic2', num_partitions=3,
                          replication_factor=2)

        admin.create_topics(new_topics=[topic, topic2])
        print("Topics created.")

    except Exception as e:
        print(e)


if __name__ == "__main__":
    main()

File: cassidy/js_code/topic.js
const { Kafka } = require("kafkajs")

run();

async function run() {
    try {
        const kafka = new Kafka({
            "clientId": "lab2",
            "brokers": ["localhost:9093", "localhost:9094", "localhost:9095"]
        });

        const admin = kafka.admin();
        console.log("Trying to connect ...");
        await admin.connect();
        console.log("Connected");

        admin.createTopics({
            "topics": [{
                "topic": "topic",
                "numPartitions": 3,
                "replicationFactor": 3
            }, {
                "topic": "topic2",
                "numPartitions": 3,
                "replicationFactor": 2
            }]
        });

        console.log("Topics created");

        await admin.disconnect();

    } catch(e) {
        console.error(e);
    } finally {
        process.exit(0);
    }
}
